### PR TITLE
used eager loading and some other methods

### PR DESCRIPTION
**Pull request by andela-oakinniranye**

### CODE REVIEW

File: app/models/concerns/votes_counter.rb [hunk 1]
```diff
@@ -16,7 +16,14 @@ module VotesCounter
   end
 
   def votes_count
-    try(:total_votes) || votes.sum(:value)
+    try(:total_votes) || votes_alternative
+  end
+
+  def votes_alternative
```
Comment (andela-ooranagwa) [inline app/models/concerns/votes_counter.rb]: I think we can remove the `:to_i` and assignment to a variable (`t`). 
```ruby
def votes_alternative
  votes.reduce(0) do |_ , v|
        _ + v
     end
end
```
... and with variable name

File: app/models/question.rb [hunk 2]
```diff
@@ -24,7 +24,9 @@ def time_updated
   end
 
   def self.with_associations
-    includes(:answers).includes(:user).includes(:comments)
+    eager_load(:votes).eager_load(answers: [{comments: [:user, :votes]}, :user, :votes]).
+    eager_load(:user).eager_load(comments: [:user, :votes]).eager_load(:tags)
+
   end
 
   def self.with_basic_association
```
Comment (andela-ooranagwa) [inline app/models/question.rb]: Any need for this method?

File: app/models/question.rb [hunk 3]
```diff
@@ -24,7 +24,9 @@ def time_updated
   end
 
   def self.with_associations
```
Comment (andela-ooranagwa) [inline app/models/question.rb]: Nice :+1:

File: app/models/concerns/votes_counter.rb (hunk 1 above)
Comment (andela-oakinniranye) [inline app/models/concerns/votes_counter.rb]: yeah, that should be fixed

File: app/models/question.rb (hunk 2 above)
Comment (andela-oakinniranye) [inline app/models/question.rb]: yeah, one is useful for the `index` and the `show`

File: app/models/concerns/votes_counter.rb [hunk 4]
```diff
@@ -16,7 +16,13 @@ module VotesCounter
...
+    votes.reduce(0) do |_ , v|
+      _.to_i + v.value
```
Comment (andela-ooranagwa) [inline app/models/concerns/votes_counter.rb]: There is really no need for the conversion to integer (`:to_i`). What do you think?
... and @andela-iamadi will **wince** if he sees these variable names :speak_no_evil:

File: app/models/question.rb (hunk 2 above)
Comment (andela-ooranagwa) [inline app/models/question.rb]: I see. You are right.

**Pull request merged**